modules/channel-vonage/src/backend/client.ts

```diff
@@ -296,9 +298,21 @@ export class VonageClient {
 
       if (picture) {
         await this.sendImage(event, { url: picture, type: 'image' })
+
+        // Sandbox API is limited to one call per second
+        if (this.config.useTestingApi === true) {
+          await Promise.delay(1000)
+        }
       }
 
       await this.sendOptions(event, body, options)
+
+      // Sandbox API is limited to one call per second. Wait one second between calls
+      if (this.config.useTestingApi === true && counter < payload.elements.length - 1) {
```

Comment:
I don't think counter logic is relevant. Being in test mode and having everything delayed 1 for 1 sec you will tolerate the last message to wait 1 sec as well no need for superfluous logic for this.

in fact, while being in testing we should always wait for 1 sec after everything that we send otherwise it might not work with some flows. let's say you have a node with 3 subsequent say instructions, if we don't delay 1 sec we'll reach the limit

Reply:
> in fact, while being in testing we should always wait for 1 sec after everything that we send otherwise it might not work with some flows. let's say you have a node with 3 subsequent say instructions, if we don't delay 1 sec we'll reach the limit

I agree, but shouldn't there be typing events in between each instruction?

Reply:
that depends if the users selects the typing indicator